awd/crossref.py

```diff
@@ -11,20 +13,18 @@
 def get_dois_from_spreadsheet(file: str) -> Iterator[str]:
     """Retriev DOIs from the Wiley-provided CSV file."""
     with smart_open.open(file, encoding="utf-8-sig") as csvfile:
-        for doi in csvfile.read().splitlines():
-            yield doi
+        yield from csvfile.read().splitlines()
 
 
 def get_work_record_from_doi(api_url: str, doi: str) -> dict[str, Any]:
-    """Retrieve Crossref works based on a DOI"""
-    work_record = requests.get(
+    """Retrieve Crossref works based on a DOI."""
+    return requests.get(
         f"{api_url}{doi}",
         params={
             "mailto": "dspace-lib@mit.edu",
         },
         timeout=30,
     ).json()
```

Comment:
Does this need error handling?  If the request is unsuccessful, or the response cannot be parsed with `.json()`, this will bubble up to the calling context which does not have any error handling either:

```python
crossref_work_record = crossref.get_work_record_from_doi(metadata_url, doi)
```

Maybe it's not required, as maybe will be handled in additional refactoring, just flagging as something to consider.

Reply:
Addressing the `json()` issue with some needed refactoring but planning a more comprehensive review of error handling and flow across the repo